File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        if ch not in word:
            return word
        else:
            start, end =0, word.find(ch)
            wordList = list(word)
            while(end > start):
                wordList[start], wordList[end] = wordList[end], wordList[start]
                start += 1
                end -= 1
                
            word = ''.join(wordList)
            return word

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.cpp
class Solution {
public:
    int find(string word, char ch){
        for(int i=0; i<word.length(); ++i){
            if(word[i] == ch)
                return i;
        }
        return -1;
    }
    string reversePrefix(string word, char ch) {
        int end = find(word, ch), start = 0;
        if(end == -1)
            return word;
        
        while(end > start){
            swap(word[start++], word[end--]);
        }
        return word;
    }
};